hana/lib/HANAService.js

```diff
@@ -837,6 +843,33 @@ class HANAService extends SQLService {
       return this.INSERT_entries(q)
     }
 
+    INSERT_select(q) {
+      const { INSERT } = q
+      const entity = this.name(q._target.name, q)
+      const alias = INSERT.into.as
+      const elements = q.elements || q._target?.elements || {}
+      const columns = (INSERT.columns || ObjectKeys(elements)).filter(
+        c => c in elements && !elements[c].virtual && !elements[c].isAssociation,
+      )
+      
+      if (columns.length !== INSERT.from.SELECT.columns.length)
+        throw new Error('The number of specified columns does not match the number of selected columns')
+      this.columns = []
+      INSERT.from.SELECT.columns = INSERT.from.SELECT.columns
+        .map((col, originalIdx) => [typeof col == 'string' ? col : this.column_name(col), originalIdx])
+        .sort(([colNameA], [colNameB]) => colNameA > colNameB ? 1 : -1)
+        .map(([, originalIdx], i) => {
+          this.columns[i] = columns[originalIdx]
+          return INSERT.from.SELECT.columns[originalIdx]
+        })
```

Comment:
Without tuples the `column_name` function would could get called more often, but it would always return the same pointer. Could look a bit cleaner by having a variable for `INSERT.from.SELECT.columns`.

```suggestion
      INSERT.from.SELECT.columns = INSERT.from.SELECT.columns
        .map((_, index) => index)
        .sort((a, b) => this.column_name(INSERT.from.SELECT.columns[a]) > this.column_name(INSERT.from.SELECT.columns[b]) ? 1 : -1)
        .map((index, i) => {
          this.columns[i] = columns[index]
          return INSERT.from.SELECT.columns[index]
        })
```